appsec/src/helper/sampler.hpp

```diff
@@ -82,10 +87,38 @@ class sampler {
         return result;
     }
 
+    static double valid_sample_rate(double sampler_rate)
+    {
+        // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)
+        if (sampler_rate <= 0) {
+            sampler_rate = 0;
+        } else if (sampler_rate > 1) {
+            sampler_rate = 1;
+        } else if (sampler_rate < min_rate) {
+            sampler_rate = min_rate;
+        }
+        // NOLINTEND(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)
+
+        return sampler_rate;
+    }
+
+    void set_sampler_rate(double sampler_rate)
```

Comment:
This should be protected, even more so when it should be used behind a mutex.

Reply:
Done